SocketTimeoutException in the Kubernetes environment.

When we keep invoking an abnormal action, it could cause the `SocketTimeoutException` in the invoker and it makes the invoker no longer able to create pods.

## Environment details:

* Kubernetes

## Steps to reproduce the issue:

1.  Create a black-box action with an abnormal image such as `centos:centos7` that does not have any kind of action proxy.
2.  Keep invoking the action every 1s.
```bash
while :
do
  wsk action invoke abnormal-image
  sleep 1s
done
```
3.  After around 1 hour, an invoker starts printing `SocketTimeoutException` even for normal actions as well.

```bash
[2021-04-05T10:38:20.573Z] [ERROR] [#tid_sid_invokerWarmup] [KubernetesClient] Failed create pod for 'wsk-blue-real-invoker-07-175029-prewarm-shellv2': class io.fabric8.kubernetes.client.KubernetesClientException (Caused by: java.net.SocketTimeoutException: timeout) - An error has occurred.; stacktrace: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:243)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:195)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:409)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:365)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:234)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:796)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:326)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:322)
	at org.apache.openwhisk.core.containerpool.kubernetes.KubernetesClient.$anonfun$run$3(KubernetesClient.scala:167)
	at scala.util.Try$.apply(Try.scala:213)
	at org.apache.openwhisk.core.containerpool.kubernetes.KubernetesClient.run(KubernetesClient.scala:166)
	at org.apache.openwhisk.core.containerpool.kubernetes.KubernetesContainer$.create(KubernetesContainer.scala:76)
	at org.apache.openwhisk.core.containerpool.kubernetes.KubernetesContainerFactory.createContainer(KubernetesContainerFactory.scala:82)
	at org.apache.openwhisk.core.invoker.InvokerReactiveV2.$anonfun$childFactory$2(InvokerReactiveV2.scala:291)
	at org.apache.openwhisk.core.invoker.InvokerReactiveV2.$anonfun$childFactory$2$adapted(InvokerReactiveV2.scala:291)
	at org.apache.openwhisk.core.containerpool.v2.ContainerProxyV2$$anonfun$1.applyOrElse(ContainerProxyV2.scala:209)
	at org.apache.openwhisk.core.containerpool.v2.ContainerProxyV2$$anonfun$1.applyOrElse(ContainerProxyV2.scala:197)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at akka.actor.FSM.processEvent(FSM.scala:707)
	at akka.actor.FSM.processEvent$(FSM.scala:704)
	at org.apache.openwhisk.core.containerpool.v2.ContainerProxyV2.processEvent(ContainerProxyV2.scala:147)
	at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:701)
	at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:695)
	at akka.actor.Actor.aroundReceive(Actor.scala:539)
	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
	at org.apache.openwhisk.core.containerpool.v2.ContainerProxyV2.aroundReceive(ContainerProxyV2.scala:147)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:612)
	at akka.actor.ActorCell.invoke(ActorCell.scala:581)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.SocketTimeoutException: timeout
	at okhttp3.internal.http2.Http2Stream$StreamTimeout.newTimeoutException(Http2Stream.java:656)
	at okhttp3.internal.http2.Http2Stream$StreamTimeout.exitAndThrowIfTimedOut(Http2Stream.java:664)
	at okhttp3.internal.http2.Http2Stream$FramingSource.read(Http2Stream.java:398)
	at okhttp3.internal.http2.Http2Codec$StreamFinishingSource.read(Http2Codec.java:205)
	at okio.RealBufferedSource$1.read(RealBufferedSource.java:439)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:233)
	... 32 more
```

This means, if any malicious user keeps trying to invoke an abnormal action, it could break the system.

## Additional information you deem important:
I confirmed it does not happen anymore after upgrading the k8s client.

--- a/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/kubernetes/KubernetesClient.scala
+++ b/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/kubernetes/KubernetesClient.scala
@@ -182,7 +182,9 @@ class KubernetesClient(
             case e =>
               transid.failed(this, start, s"Failed create pod for '$name': ${e.getClass} - ${e.getMessage}", ErrorLevel)
               //log pod events to diagnose pod readiness failures
-              val podEvents = kubeRestClient.events
+              val podEvents = kubeRestClient
+                .v1()
+                .events()
                 .inNamespace(namespace)
                 .withField("involvedObject.name", name)
                 .list()

--- a/settings.gradle
+++ b/settings.gradle
@@ -67,4 +67,4 @@ gradle.ext.akka_http = [version : '10.2.4']
 gradle.ext.akka_management = [version : '1.0.5']
 
 gradle.ext.curator = [version : '4.0.0']
-gradle.ext.kube_client = [version: '4.4.2']
+gradle.ext.kube_client = [version: '4.10.3']

--- a/tests/src/test/scala/org/apache/openwhisk/core/containerpool/kubernetes/test/WhiskPodBuilderTests.scala
+++ b/tests/src/test/scala/org/apache/openwhisk/core/containerpool/kubernetes/test/WhiskPodBuilderTests.scala
@@ -88,21 +88,21 @@ class WhiskPodBuilderTests extends FlatSpec with Matchers with KubeClientSupport
     withClue(Serialization.asYaml(pod)) {
       val c = getActionContainer(pod)
       //min cpu is: config.millicpus
-      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("300m")
+      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("300")
     }
 
     val (pod2, _) = builder.buildPodSpec(name, testImage, 15.MB, Map("foo" -> "bar"), Map("fooL" -> "barV"), config)
     withClue(Serialization.asYaml(pod2)) {
       val c = getActionContainer(pod2)
       //max cpu is: config.maxMillicpus
-      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("1000m")
+      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("1000")
     }
 
     val (pod3, _) = builder.buildPodSpec(name, testImage, 7.MB, Map("foo" -> "bar"), Map("fooL" -> "barV"), config)
     withClue(Serialization.asYaml(pod3)) {
       val c = getActionContainer(pod3)
       //scaled cpu is: action mem/config.mem x config.maxMillicpus
-      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("600m")
+      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("600")
     }
 
     val config2 = KubernetesClientConfig(
@@ -139,8 +139,8 @@ class WhiskPodBuilderTests extends FlatSpec with Matchers with KubeClientSupport
     val (pod, _) = builder.buildPodSpec(name, testImage, 2.MB, Map("foo" -> "bar"), Map("fooL" -> "barV"), config)
     withClue(Serialization.asYaml(pod)) {
       val c = getActionContainer(pod)
-      c.getResources.getLimits.asScala.get("ephemeral-storage").map(_.getAmount) shouldBe Some("1024Mi")
-      c.getResources.getRequests.asScala.get("ephemeral-storage").map(_.getAmount) shouldBe Some("1024Mi")
+      c.getResources.getLimits.asScala.get("ephemeral-storage").map(_.getAmount) shouldBe Some("1024")
+      c.getResources.getRequests.asScala.get("ephemeral-storage").map(_.getAmount) shouldBe Some("1024")
     }
   }
 
@@ -222,8 +222,8 @@ class WhiskPodBuilderTests extends FlatSpec with Matchers with KubeClientSupport
         new EnvVar("foo", "bar", null),
         new EnvVar("POD_UID", null, new EnvVarSource(null, new ObjectFieldSelector(null, "metadata.uid"), null, null))))
 
-      c.getResources.getLimits.asScala.get("memory").map(_.getAmount) shouldBe Some("10Mi")
-      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("900m")
+      c.getResources.getLimits.asScala.get("memory").map(_.getAmount) shouldBe Some("10")
+      c.getResources.getLimits.asScala.get("cpu").map(_.getAmount) shouldBe Some("900")
       c.getSecurityContext.getCapabilities.getDrop.asScala should contain allOf ("NET_RAW", "NET_ADMIN")
       c.getPorts.asScala.find(_.getName == "action").map(_.getContainerPort) shouldBe Some(8080)
       c.getImage shouldBe testImage

